### PR TITLE
Authen

### PR DESCRIPTION
**Pull request by georgezheng999**
This PR covers the integration of the Authenticator library. 

Additional features incorporated include: association of comments to email addresses and establishing a login/logout system.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    if (userService.isUserLoggedIn()) {
+      final String userEmail = userService.getCurrentUser().getEmail();
+      final String urlToRedirectToAfterUserLogsOut = "/index.html";
+      final String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+
+      response.getWriter().println("<p>Hello " + userEmail + "!</p>");
+      response.getWriter().println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
+    } else {
+      final String urlToRedirectToAfterUserLogsIn = "/index.html";
+      final String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
```
Comment (qlyao) [inline portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:39]: Can we just redirect the user to the loginUrl? It looks a little weird when a user comes to your "login" endpoint but need to click an extra button to log in.

File: portfolio/src/main/webapp/index.html [hunk 2]
```diff
@@ -6,70 +6,24 @@
...
-    </div>
-    <div id="background">
-      <p>I have worked on a variety of projects spanning several academic disciplines (check out some selected works featured below!). In my free time, I enjoy playing video games (most recently, League of Legends and Realm of the Mad God), watching anime, and napping. 
+  <body onload="handleLogin()">
+    <div id="commenting">
+      <button type="button" onclick="deleteComments()">Delete Comments</button>
+      <ul id="comments-history"></ul>
       <form>
-        <input type="button" value="Select a project!" onClick="chooseFeaturedProj()">
+        <p>Enter number of comments to view:</p>
```
Comment (qlyao) [inline portfolio/src/main/webapp/index.html]: Consider making it a `<label>` that is associated with the `<input>` below.

FYI, https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java (hunk 1 above)
Comment (georgezheng999) [reply portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:39]: done.

File: portfolio/src/main/webapp/index.html (hunk 2 above)
Comment (georgezheng999) [reply portfolio/src/main/webapp/index.html]: changed

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html");
+    final UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      final String userEmail = userService.getCurrentUser().getEmail();
+      final String urlToRedirectToAfterUserLogsOut = "/index.html";
+      final String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+      response.getWriter().println("<p>Hello " + userEmail + "!</p>");
+      response.getWriter().println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
```
Comment (ChuxiangP) [inline portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java]: Maybe do the same for the log out flow (remove extra click)?

**Review by ChuxiangP — APPROVED** [review]

**Review by qlyao — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java (hunk 3 above)
Comment (georgezheng999) [reply portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java]: Yes

**Pull request merged**